### PR TITLE
Problem: potentially unneeded, deprecated and legacy functions in use

### LINKED ISSUE #338: Problem: potentially unneeded, deprecated and legacy functions in use
> A number of deprecated and legacy functions are still in use throughout the codebase.

> AminoCodec in the file x/supply/types/codec.go (this also applies to the file x/chainmain/types/codec.go). These are called by the method RegisterLegacyAminoCodec of AppModuleBasic in x/supply/module.go. Similarly, consider removing functions that are not used, such as the method LegacyQuerierHandler() of AppModule defined in x/chainmain/module.go. Most of the functions in x/chainmain/module.go are either empty, or perform default behavior. Similarly, the function InitGenesis() in x/chainmain/genesis.go is empty and the function ExportGenesis() of the same file returns the default genesis. The function RegisterRoutes() in x/chainmain/client/rest/rest.go is also empty.

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Removed legacy API from supply module and unused code from chainmain module. Fixes #338.

### CODE REVIEW

File: x/supply/keeper/querier.go [hunk 1]
```diff
@@ -1,56 +0,0 @@
...
-	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
-	"github.com/crypto-com/chain-main/x/supply/types"
-	abci "github.com/tendermint/tendermint/abci/types"
-)
-
-// NewQuerier returns a new sdk.Keeper instance.
-func NewQuerier(k Keeper, legacyQuerierCdc *codec.LegacyAmino) sdk.Querier {
-	return func(ctx sdk.Context, path []string, req abci.RequestQuery) ([]byte, error) {
-		switch path[0] {
-		case types.QueryTotalSupply:
```
Comment (yihuang) [inline x/supply/keeper/querier.go:15]: Are these queriers not used?

Comment (devashishdxt) [reply x/supply/keeper/querier.go:15]: I don't think so. @calvinaco can clarify.

Comment (tomtau) [reply x/supply/keeper/querier.go:15]: checked with @calvinaco -- this legacy one wasn't being used or planned to be used by third parties

**Review by tomtau — APPROVED** [review]

**Pull request merged**